src/runtime/lpython_builtin.py

```diff
@@ -183,3 +183,43 @@ def round(value: f64) -> i32:
         return int(value)
     else:
         return int(value) + 1
+
+
+def _partition(array: list[f64], start: i32, end: i32) -> i32:
+    pivot: f64
+    pivot = array[start]
+    low: i32
+    low = start + 1
+    high: i32
+    high = end
+    tmp: f64
+
+    while True:
+        while low <= high and array[high] >= pivot:
+            high = high - 1
+        while low <= high and array[low] <= pivot:
+            low = low + 1
+        if low <= high:
+            tmp = array[low]
+            array[low] = array[high]
+            array[high] = tmp
+        else:
+            break
+    tmp = array[start]
+    array[start] = array[high]
+    array[high] = tmp
+
+    return high
+
+
+def quick_sort(array: list[f64], start: i32, end: i32):
+    if start >= end:
+        return
+    p: i32
+    p = _partition(array, start, end)
+    quick_sort(array, start, p-1)
+    quick_sort(array, p+1, end)
+
+
+def sorted(array: list[f64]):
+    quick_sort(array, 0, len(array)-1)
```

Comment:
Well Python uses [Tim sort](https://en.wikipedia.org/wiki/Timsort), a combination of merge and insertion sort. Quick sort's time complexity depends on pivot selection and the ordering of input elements. Merge sort's time complexity is stable in that sense. So, I believe there must be good reasons to use quick sort and not Tim sort here.

Reply:
In addition I think this recursive implementation is limiting (due to the maximum number of recursions one can do on a system). For large arrays this will fail. For small arrays, insertion sort should be faster anyways. So my suggestion - Quick sort (or a better choice Tim sort) can be implemented recursively. See, https://github.com/codezonediitj/pydatastructs/blob/3abc0264cb97fcb90b82e95eed4b1bc6b1148049/pydatastructs/linear_data_structures/algorithms.py#L751-L790 for example.

Reply:
I think `quick_sort` is easy to implement, so we can start with that. I think Timsort is better, but much harder to implement. Once we have it, we should switch it.

Reply:
Sure. @Smit-create Is there any blocker which is preventing from implementing this feature? Or is it low on the priority list?